tools/firely-auth/firely-auth-settings.rst

```diff
@@ -0,0 +1,299 @@
+- ``appsettings.instance.json`` - This file is meant to override settings for this instance of Firely Auth. You can create this file yourself by copying (parts of) ``appsettings.default.json``.
+- Environment variables with the prefix ``FIRELY_AUTH_`` - Use environment variables to more easily configure settings from CI/CD pipelines, secure vaults etc.
+
+Unsure how to name your variables? This works the same as with :ref:`configure_envvar`.
+
+Sections
+--------
+
+.. _firely_auth_settings_license:
+
+License
+^^^^^^^
+
+Use the License settings to set the location to the license file. A relative path is evaluated relative to the executable ``Firely.IdentityServer.Core.exe``.
+You can use the same license file that came with Firely Server.
+
+.. code-block:: json
+
+  "License": {
+    "LicenseFile": "./firelyserver-license.json"
+  },
+
+
+.. _firely_auth_settings_kestrel:
+
+Kestrel
+^^^^^^^
+
+Kestrel is the web server that is integrated into Firely Auth, just like for Firely Server. We recommend to run both behind a :ref:`reverse proxy <deploy_reverseProxy>`.
+Nevertheless you can control the settings for Kestrel.
+
+.. code-block:: json
+    
+  "Kestrel": {
+    "Endpoints": {
+      "Http": {
+        "Url": "http://localhost:5100"
+      },
+      "HttpsFromPem": {
+        "Url": "https://localhost:5101",
+        "SslProtocols": [ "Tls12", "Tls13" ],
+        "Certificate": {
+          "Path": "cert.pem",
+          "KeyPath": "cert-key.pem"
+        }
+      }
+    }
+   },
+ 
+These settings are not Firely Auth specific, and you can read more about them in the `Microsoft documentation <https://docs.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints>`_.
+In that documentation you can also read how to use the ``Https`` setting instead of ``HttpsFromPem`` to use a ``.pfx`` file for your SSL certificate.
+
+.. _firely_auth_settings_server:
+
+Firely Server
+^^^^^^^^^^^^^
+
+Firely Auth hands out SMART on FHIR access tokens to access resources on Firely Server. 
+To make Firely Server known to Firely Auth, fill in the ``FhirServerConfig``:
+
+.. code-block:: json
+
+   "FhirServerConfig": {
+      "Name": "Firely Server",
+      "FHIR_BASE_URL": "http://localhost:4080"
+   },
+
+- ``Name``: This name serves two purposes:
+
+  - It can be added to the token as the value of the ``aud`` (audience) claim, if the client requests so. 
+    To have it accepted by Firely Server, set its ``SmartAuthorizationOptions:Audience`` setting to the same value.
+    (This currently only works for the Authorization Code flow.)
+  - It correlates with the clients allowed to access the token introspection endpoint (see below).
+    Therefore it should match the value of ``TokenIntrospectionConfig:TokenIntrospectionResources:Name``
+
+- ``FHIR_BASE_URL``: A token can have a claim in the form of ``patient=<base>/Patient/123``, to define the compartment the client is restricted to.
```

Comment:
It is also the url against which the aud parameter in a token request is checked

Reply:
The 'aud' parameter is set to the Name, right? So `Firely Server`, not this url?

Reply:
AudienceAuthorizeRequestValidator checks this against url (for the EHR launch workflow)

Reply:
See http://hl7.org/fhir/smart-app-launch/app-launch.html#request-4 for how it's different compared to the "aud" claim

Reply:
Done. Also included example of how to add `aud` parameter to authorize request.